File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/scheduler/SchedulerXmlCommandExecutor.java
package com.sos.scheduler.engine.kernel.scheduler;

import com.google.inject.ImplementedBy;
import com.sos.scheduler.engine.kernel.security.SchedulerSecurityLevel;
import com.sos.scheduler.engine.kernel.Scheduler;

@ImplementedBy(Scheduler.class)
public interface SchedulerXmlCommandExecutor {
    String uncheckedExecuteXml(String xml);
    String uncheckedExecuteXml(String xml, SchedulerSecurityLevel secLevel);
    String executeXml(String xml);
}

File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/scheduler/SchedulerXmlCommandExecutor.scala
package com.sos.scheduler.engine.kernel.scheduler

import com.google.inject.ImplementedBy
import com.sos.scheduler.engine.kernel.Scheduler
import com.sos.scheduler.engine.kernel.security.SchedulerSecurityLevel

@ImplementedBy(classOf[Scheduler])
trait SchedulerXmlCommandExecutor {

  def uncheckedExecuteXml(xml: String): String

  def uncheckedExecuteXml(xml: String, securityLevel: SchedulerSecurityLevel): String

  def executeXml(xml: String): String
}

